File: smalivm/src/main/java/org/cf/smalivm/opcode/ExecutionContextOp.java
package org.cf.smalivm.opcode;

import org.cf.smalivm.context.ExecutionContext;
import org.cf.smalivm.context.ExecutionNode;
import org.jf.dexlib2.builder.MethodLocation;

public abstract class ExecutionContextOp extends Op {

    ExecutionContextOp(MethodLocation location, MethodLocation child) {
        super(location, child);
    }

    ExecutionContextOp(MethodLocation location, MethodLocation[] children) {
        super(location, children);
    }

    public abstract void execute(ExecutionNode executionNode, ExecutionContext context);

}

File: smalivm/src/main/java/org/cf/smalivm/opcode/ExecutionContextOp.kt
package org.cf.smalivm.opcode

import org.cf.smalivm.context.ExecutionContext
import org.cf.smalivm.context.ExecutionNode
import org.jf.dexlib2.builder.MethodLocation

abstract class ExecutionContextOp : Op {
    internal constructor(location: MethodLocation, child: MethodLocation?) : super(location, child) {}
    internal constructor(location: MethodLocation, children: Array<MethodLocation?>?) : super(location, children) {}

    abstract fun execute(executionNode: ExecutionNode?, context: ExecutionContext)
}
